lib/pokerscore/hand.rb

```diff
@@ -1,83 +1,48 @@
-require './const/hand_constants.rb'
+require 'const/hand_constants.rb'
+require 'lib/pokerscore/set.rb'
 
 class Hand
-  include Comparable
-  attr_reader :cards, :straight_high_card, :sets, :bestcards
+  attr_reader :cards
+  WHEEL_STRAIGHT = [2,3,4,5,14]
 
-  def <=>(hand2)
-    if bestcards != hand2.bestcards
-      return $hand_types[bestcards] <=> $hand_types[hand2.bestcards]
-    end
-    if straight_high_card != nil
-      return straight_high_card <=> hand2.straight_high_card 
-    end
-    values = []
-    hand2_values = []
-    (1..4).each { |index| values += sets[index].sort }
-    (1..4).each { |index| hand2_values += hand2.sets[index].sort }
-    values.zip(hand2_values).each do |value, hand2_value|
-      return value <=> hand2_value if value != hand2_value
+  def initialize(cards)
+    raise ArgumentError if invalid_hand?(cards)
+    @cards = cards
+  end
+  
+  def sets
+    sets = []
+    unique_values = cards.map(&:value).uniq
+    unique_values.each do |unique_value|
+      occurrances = cards.count { |card| card.value == unique_value }
+      sets << Set.new(unique_value, occurrances)
     end
+    sets
   end
 
-  def to_s 
-    cards.map(&:to_s).join(", ")
+  def max_value
+    cards.max { |card1, card2| card1.value <=> card2.value }.value
   end
   
-  def initialize(cards)
-    raise ArgumentError if invalid_hand?(cards)
-    @cards = cards
-    @straight_high_card = find_straight(cards)
-    @sets = find_sets(cards)
-    @bestcards = find_best_cards(cards)
+  def straight?
+    sorted_values = cards.map(&:value).sort
+    sorted_values.each_cons(2).each do |value, next_value|
+      return false if not value == next_value - 1
+    end
+    return true if sorted_values == WHEEL_STRAIGHT
```

Comment:
This line won't change our outcome. Regardless of what `sorted_values == WHEEL_STRAIGHT` evaluates to, we'll be returning true. We'll actually fail out before we reach this line, as [2,3,4,5,14] won't be accepted by the check in the loop.